test/runner.rb

```diff
@@ -1,11 +1,20 @@
 # frozen_string_literal: true
 
+require 'rbconfig'
+
 # Should be done in rubygems test files?
 ENV["GEM_SKIP"] = ENV["GEM_HOME"] = ENV["GEM_PATH"] = "".freeze
 
-# Get bundled gems on load path
-Dir.glob("#{__dir__}/../gems/*/*.gemspec")
-  .reject {|f| f =~ /minitest|test-unit|power_assert/ }
-  .map {|f| $LOAD_PATH.unshift File.join(File.dirname(f), "lib") }
+# Get bundled gems on load path, try install dir first
+gem_dir = "#{RbConfig::CONFIG['rubylibdir'].sub('/ruby/', '/ruby/gems/')}/gems"
```

Comment:
How about changing:
```ruby
"#{RbConfig::CONFIG['rubylibdir'].sub('/ruby/', '/ruby/gems/')}/gems"
```
to:
```ruby
"#{RbConfig::CONFIG['rubylibdir'].sub('/lib/ruby/', '/lib/ruby/gems/')}/gems"
```
It picks up the ABI with check RbCongi for it.  IOW, trying to change
```
/lib/ruby/2.7.0
```
to
```
/lib/ruby/gems/2.7.0
```

Reply:
When `prefix` is `/lib/ruby`, it will result in an unexpected path.
`RbConfig::CONFIG.values_at("rubylibprefix", "ruby_version").join("/gems/")` or `RbConfig.expand("$(rubylibprefix)/gems/$(ruby_version)")` would work.

But first, tests need to run in a clean environment, already installed gems are not welcome, I think.